lms/djangoapps/course_structure_api/v0/tests.py

```diff
@@ -0,0 +1,350 @@
+    Mixin for course view tests.
+    """
+    view = None
+
+    def setUp(self):
+        super(CourseViewTestsMixin, self).setUp()
+        self.create_test_data()
+        self.create_user_and_access_token()
+
+    def create_user_and_access_token(self):
+        self.user = GlobalStaffFactory.create()
+        self.oauth_client = ClientFactory.create()
+        self.access_token = AccessTokenFactory.create(user=self.user, client=self.oauth_client).token
+
+    def create_test_data(self):
+        self.invalid_course_id = 'foo/bar/baz'
+        self.course = CourseFactory.create(display_name='An Introduction to API Testing', raw_grader=[
+            {
+                "min_count": 24,
+                "weight": 0.2,
+                "type": "Homework",
+                "drop_count": 0,
+                "short_label": "HW"
+            },
+            {
+                "min_count": 4,
+                "weight": 0.8,
+                "type": "Exam",
+                "drop_count": 0,
+                "short_label": "Exam"
+            }
+        ])
+        self.course_id = unicode(self.course.id)
+
+        sequential = ItemFactory.create(
+            category="sequential",
+            parent_location=self.course.location,
+            display_name="Lesson 1",
+            format="Homework",
+            graded=True
+        )
+
+        ItemFactory.create(
+            category="problem",
+            parent_location=sequential.location,
+            display_name="Problem 1",
+            format="Homework"
+        )
+
+        self.empty_course = CourseFactory.create(
+            start=datetime(2014, 6, 16, 14, 30),
+            end=datetime(2015, 1, 16),
+            org="MTD"
+        )
+
+    def build_absolute_url(self, path=None):
+        """ Build absolute URL pointing to test server.
+        :param path: Path to append to the URL
+        """
+        url = TEST_SERVER_HOST
+
+        if path:
+            url += path
+
+        return url
+
+    def assertValidResponseCourse(self, data, course):
+        """ Determines if the given response data (dict) matches the specified course. """
+
+        course_key = course.id
+        self.assertEqual(data['id'], unicode(course_key))
+        self.assertEqual(data['name'], course.display_name)
+        self.assertEqual(data['course'], course_key.course)
+        self.assertEqual(data['org'], course_key.org)
+        self.assertEqual(data['run'], course_key.run)
+
+        uri = self.build_absolute_url(
+            reverse('course_structure_api:v0:detail', kwargs={'course_id': unicode(course_key)}))
+        self.assertEqual(data['uri'], uri)
+
+    def http_get(self, uri, **headers):
+        """Submit an HTTP GET request"""
+
+        default_headers = {
+            'HTTP_AUTHORIZATION': 'Bearer ' + self.access_token
+        }
+        default_headers.update(headers)
+
+        response = self.client.get(uri, content_type='application/json', follow=True, **default_headers)
```

Comment:
The content type needs to be specified to avoid returning the HTML for the API discovery pages.